_posts/2024-02-21-securing-wildfly-apps-okta-openshift.adoc

```diff
@@ -0,0 +1,260 @@
+```
+Alternatively, you can log in using the API token:
+```
+oc login --token=myToken --server=myServerUrl
+```
+You can request the token via the *Copy Login Command* link in the OpenShift web console.
+
+If you don’t already have a project created, you can create one using:
+```
+oc new-project myProjectName
+```
+
+Once you have your environment set up with the required tools, we can move on to the next step to build and deploy our application on OpenShift.
+
+== Configure Okta 
+For this guide, we will be using Okta as out OpenID provider. In order to secure the application, we will need to access the Okta web console and register our application as a client. 
+
+. Log in to the https://www.okta.com/free-trial/[Okta dashboard] to get started. 
+. At the top right corner, click on *Admin* to be redirected to the admin dashboard. 
+. On the left hand panel, from the *Applications* drop down, click on *Applications* and then *Create App Integration*. Choose *OIDC - OpenID Connect* under *Sign-in method* and *Web Application* for *Application type* and click *Next*. 
+. Under *General Settings* add the following configurations: 
+* `App integration name:` *simple-webapp-okta*
+* `Grant type:` Leave it as is, with `Authorization Code` checked. 
+* `Sign-in redirect URIs`: Leave it as is for now. We will edit it later.
+* Under the `Assignments` tab, for `Controlled access`, select *Skip group assignment for now*. 
+* Click `Save`
+. Navigate to the *Directory* drop down on the left panel and click *People*. Click *Add person* and add a person with: 
+* First name: *Alice*
+* Last name: *Smith*
+* Email: *alice@example.org*
+* User Name: *alice@example.org* 
+* From the dropdown list for _Password_, choose _Set by admin_ and set a password that fits the password requirements. 
+* Uncheck _User must change password on first login_ and hit `Save`. For more information about how to add a user manually https://help.okta.com/en-us/content/topics/users-groups-profiles/usgp-add-users.htm[click here].
+. Click on the _Applications_ tab on the left hand side. Navigate to _Applications_ and click on _simple-webapp-okta_. Under the _Assignments_ tab, click on *Assign*, then click on *Assign to People* and select *Alice*. For more information about how to assign a user to an application, please see https://support.okta.com/help/s/article/How-To-Assign-An-User-To-An-Application?language=en_US[here].
+
+=== Advanced OpenID Configuration
+In order to extract the user information through the id token, we will need to configure a custom authorization server. Follow the steps below for this:
+
+. Under the *Security* dropdown, select *API* and click on the edit button next to the *Default* server. 
+. Under the *Claims* tab, click on the *Add Claim* button and to add a claim with the following configurations: 
+* Name: *IDTClaim*
+* Include in token type: *ID Token* and keep it at always. 
+* Value type: *Expression*
+* Value: *appuser.userName* and hit `Save`. 
```

Comment:
Might be good to add a quick sentence here that says you'll be using this claim to get the username of the logged in user.

Reply:
fixed!